parser/impl/parser.cpp

```diff
@@ -240,28 +240,115 @@ sPtr<ast::RecordType> Parser::parseRecordType() {
+        if (token.type != TokenType::Var) {
+            m_errors.push_back(Error{
+                .pos = token.pos,
+                .message = "Expected \"var\" keyword but didn't find it.",
+            });
+        }
+        token = skipWhile(isNewLine);
+        if (token.type != TokenType::Identifier) {
+            m_errors.push_back(Error{
+                .pos = token.pos,
+                .message = "Expected an identifier after \"var\" keyword.",
+            });
+        }
+	    variable.name = token;
+        token = skipWhile(isNewLine);
+        if (token.type != TokenType::Is && token.type != TokenType::Colon) {
+            m_errors.push_back(Error{
+                .pos = token.pos,
+                .message = "Expected an \"is\" keyword or : after the identifier.",
+            });
+        }
+        switch (token.type)
+        {
+	        case TokenType::Colon:
+                variable.type = parseType();
+                if ((token = m_lexer.Peek()).type != TokenType::Is) break;
+		        else m_lexer.Next();
+            case TokenType::Is:
+                variable.expression = parseExpression();
+                break;
+            default:
+                m_errors.push_back(Error{
+                    .pos = currentToken.pos,
+                    .message = "Expected an \"is\" keyword or : after the identifier.",
+                });
+        }
+        return std::make_shared<ast::Variable>(variable);
+  }        
+  
+   sPtr<ast::Body> Parser::parseBody() {
+	ast::Body body;
+        auto currentToken;
+        while ((currentToken = m_lexer.Peek()).type != TokenType::End)
+        {
+            switch (currentToken.type)
+            {
+            case TokenType::Var:
+                body.variables.push_back(parseVariable());
+                continue;
+            case TokenType::Type:
+                body.types.push_back(parseType());
+                continue;
+            case TokenType::NewLine:
+                m_lexer.Next();
+                continue;
+            default:
+                body.statements.push_back(parseStatement());
+            }
+        }
+        return std::make_shared<ast::Body>(body);
+    }
+
+    sPtr<ast::Statement> Parser::parseStatement() {
+        auto currentToken = skipWhile(isNewLine);
+        switch (currentToken.type)
+        {
+        case TokenType::Identifier:
+	        sPtr<Expression> expression = parseExpression();
+	        if (dynamic_cast<sPtr<ast::RoutineCall>>(expression))
+	            return expression;
+            else 
+                return parseAssignment(expression);
+        case TokenType::While:
+            return parseWhileLoop();
+        case TokenType::For:
+            return parseForLoop();
+	    case TokenType::If:
+            return parseIfStatement();
+        default:
+            m_errors.push_back(Error{
+                    .pos = currentToken.pos,
+                    .message = "Unexpected token",
+                });
```

Comment:
What's up with the indentation?

Reply:
It didnt merge automated (i dont know why), I did it by hand